### PR TITLE
add 3D batchnorm tests

### PR DESCRIPTION
**Pull request by dnikolaev-amd**
3D batchhorm tests (NHWC3D and NCHW3D)

NCHW 3D tests:

```
python test_nn.py -v -k test_batchnorm_train_NCHW3D_vs_native_float32
python test_nn.py -v -k test_batchnorm_train_NCHW3D_vs_native_mixed_float16
python test_nn.py -v -k test_batchnorm_train_NCHW3D_vs_native_mixed_bfloat16

python test_nn.py -v -k test_batchnorm_inference_NCHW3D_vs_native_float32
python test_nn.py -v -k test_batchnorm_inference_NCHW3D_vs_native_mixed_float16
python test_nn.py -v -k test_batchnorm_inference_NCHW3D_vs_native_mixed_bfloat16
```
NHWC 3D tests:
```
python test_nn.py -v -k test_batchnorm_train_NHWC3D_vs_native_float32
python test_nn.py -v -k test_batchnorm_train_NHWC3D_vs_native_mixed_float16
python test_nn.py -v -k test_batchnorm_train_NHWC3D_vs_native_mixed_bfloat16

python test_nn.py -v -k test_batchnorm_inference_NHWC3D_vs_native_float32
python test_nn.py -v -k test_batchnorm_inference_NHWC3D_vs_native_mixed_float16
python test_nn.py -v -k test_batchnorm_inference_NHWC3D_vs_native_mixed_bfloat16
```

### CODE REVIEW

**Review by jithunnair-amd — APPROVED** [review]

File: test/test_nn.py [hunk 1]
```diff
@@ -5123,21 +5121,37 @@ def test_batchnorm_nhwc_cuda(self):
...
         name_fn=lambda f, b, m, t: f"{f}_vs_{b}{'_mixed' if m else ''}_{dtype_name(t)}"
     )
-    def test_batchnorm(self, mode, memory_format, ref_backend, mixed, dtype):
+    def test_batchnorm(self, dims, mode, memory_format, ref_backend, mixed, dtype):    
+        if torch.version.hip:
+            if self._testMethodName in ("test_batchnorm_2D_train_NHWC_vs_NCHW_mixed_bfloat16", 
+                                    "test_batchnorm_2D_train_NHWC_vs_cpu_mixed_bfloat16",
+                                    "test_batchnorm_3D_train_NHWC_vs_NCHW_mixed_bfloat16", 
+                                    "test_batchnorm_3D_train_NHWC_vs_cpu_mixed_bfloat16"
+                                    ) and _get_torch_rocm_version() < (6, 4):
```
Comment (jithunnair-amd) [inline test/test_nn.py:5134]:
```suggestion
            if self._testMethodName in ("test_batchnorm_2D_train_NHWC_vs_NCHW_mixed_bfloat16", 
                                    "test_batchnorm_2D_train_NCHW_vs_cpu_mixed_bfloat16",
                                    "test_batchnorm_3D_train_NHWC_vs_NCHW_mixed_bfloat16", 
                                    "test_batchnorm_3D_train_NCHW_vs_cpu_mixed_bfloat16"
                                    ) and _get_torch_rocm_version() < (6, 4):
```

typo?

File: test/test_nn.py [hunk 2]
```diff
@@ -5123,21 +5121,37 @@ def test_batchnorm_nhwc_cuda(self):
...
+                # NCHW bfloat16 path uses native kernels for rocm<=6.3
+                # train failed on rocm<=6.3 due to native tolerance issue SWDEV-507600
+                self.skipTest("bfloat16 NHWC train failed on ROCm <= 6.3")
+
+            if self._testMethodName in ("test_batchnorm_2D_train_NCHW_vs_native_mixed_bfloat16",
+                                        "test_batchnorm_3D_train_NCHW_vs_native_mixed_bfloat16"
+                                        ):
+                self.skipTest("bfloat16 NHWC train failed due to native tolerance issue SWDEV-507600")
```
Comment (jithunnair-amd) [inline test/test_nn.py:5142]:
```suggestion
            if self._testMethodName in ("test_batchnorm_2D_train_NCHW_vs_native_mixed_bfloat16",
                                        "test_batchnorm_3D_train_NCHW_vs_native_mixed_bfloat16"
                                        ) and _get_torch_rocm_version() >= (6, 4)::
                self.skipTest("bfloat16 NCHW train failed due to native tolerance issue SWDEV-507600")
```

**Review by jithunnair-amd — CHANGES_REQUESTED** [review]
Can you please make the corrections based on the comments and then run all the tests on a 6.3 and 6.4+ docker image to ensure that we got the skip logic right?

**Review by jithunnair-amd — APPROVED** [review]

**Comment by jithunnair-amd** [comment]
!cherry-pick --onto release/2.7 rocm7.0_internal_testing

**Comment by okakarpa** [comment]
Created branch [autogenerated/release/2.7_cherry-pick_pr-2214](https://github.com/ROCm/pytorch/tree/autogenerated/release/2.7_cherry-pick_pr-2214) and https://github.com/ROCm/pytorch/pull/2243

Created branch [autogenerated/rocm7.0_internal_testing_cherry-pick_pr-2214](https://github.com/ROCm/pytorch/tree/autogenerated/rocm7.0_internal_testing_cherry-pick_pr-2214) and https://github.com/ROCm/pytorch/pull/2244. It contains a merge conflict. Please resolve it
